theano/scan_module/scan_utils.py

```diff
@@ -447,21 +449,51 @@ def equal_computations(xs, ys, in_xs=None, in_ys=None):
         elif len(nd_x.outputs) != len(nd_y.outputs):
             return False
         else:
```

Comment:
Should we check if the outputs are all already in `common` first?
It could happen if we already encountered the node in a previous recursion.